tools/autograd/derivatives.yaml

```diff
@@ -673,6 +673,16 @@
   self: not_implemented("lstsq")
   A: not_implemented("lstsq")
 
+  # we mark linalg_lstsq non_differentiable wrt to its Tensor inputs because,
+  # if marked not_implemented, an assertion is triggered in
+  # torch/csrc/autograd/utils.h:64 which suggests adding the function
+  # to the DONT_REQUIRE_DERIVATIVE list in tools/autograd/gen_variable_type.py,
+  # resulting in essentially the same goal, i.e. producing an output with
+  # requires_grad=False
+- name: linalg_lstsq(Tensor self, Tensor b, float? cond=None, *, str? driver=None) -> (Tensor solution, Tensor residuals, Tensor rank, Tensor singular_values)
+  self: non_differentiable
```

Comment:
What you want to set here I think is `output_differentiability` that way you can specify for each output if it is differentiable or not. All the ones that you mark False will always have `requires_grad=False`.

Or do you want another behavior?

Reply:
Tried `output_differentiability` but it gives the same result. My goal is to get an error explaining that gradient is not implemented. I cannot achieve this by setting `not_implemented` for this linalg function.

Reply:
For example, suppose I set `self/b: not_implemented('linalg_lstsq')`, then at runtime:
```python
In [1]: import torch
x
In [2]: x = torch.rand(3, 3).requires_grad_(True)

In [3]: b = torch.rand(3, 3).requires_grad_(True)

In [4]: torch.linalg.lstsq(x, b)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-4-439eae9f5368> in <module>
----> 1 torch.linalg.lstsq(x, b)

RuntimeError: isDifferentiableType(variable.scalar_type()) INTERNAL ASSERT FAILED at "../torch/csrc/autograd/functions/utils.h":64, please report a bug to PyTorch. 
```

Reply:
@albanD , any suggestions? Shall I keep it the way it is already in the code?

Reply:
Sorry I missed your question!
> Tried output_differentiability but it gives the same result.
What behavior is that?

Also if you want some outputs to not require gradients and some to do but raise an error if you try to backward, you can use output_differentiability to mark the ones that should not require gradients and define `  self: not_implemented("lstsq with respect to self:)` for the ones that you marked as differentiable.

Reply:
Problem solved, thank you, @albanD !